index.js

```diff
@@ -19,48 +16,92 @@ module.exports = (input, options) => {
   options = Object.assign({}, options)
 
   let table = ''
-  let titles = []
-  let alignments = []
 
   let keys = Object.keys(input[0])
 
-  if (!options.columns || !Array.isArray(options.columns)) {
-    // use the keys from the first object
-    titles = keys
-    alignments = new Array(titles.length).fill('')
-  } else {
-    titles = keys.map((key, i) => {
-      let column = options.columns[i]
-
-      if (isPlainObject(column)) {
-        alignments[i] = column.align
-        return column.name || key
-      } else {
-        alignments[i] = ''
-        return column || key
-      }
-    })
-  }
+  let titles = keys.map((key, i) => {
+    if (Array.isArray(options.columns) && options.columns[i]) {
+        if (typeof options.columns[i] === 'string') {
+          return options.columns[i]
+        } else if (options.columns[i].name) {
+          return options.columns[i].name
+        }
+    }
+
+    return key
+  })
+
+  let widths = input.reduce(
+    (sizes, item) => keys.map(
+      (key, i) => Math.max(
+        columnWidthMin,
+        typeof item[key] === 'undefined' ? 0 : String(item[key]).length,
+        sizes[i]
+      )
+    ),
+    titles.map(t => t.length)
+  )
+
+  let alignments = keys.map((key, i) => {
+    if (Array.isArray(options.columns)
+      && options.columns[i]
```

Comment:
Probably would be good to pull up that `options.columns[i]` value into a variable if we're using it multiple times like this.

Reply:
Which style would you prefer then? Nested `if`s or "return early"?

Reply:
I go for returning early, but I wouldn't say the code you're replacing exemplified that at all :man_facepalming: I'm actually fine with this the way you've got it.

Reply:
I usually prefer "return early", and started this code path as such, but it's twice the lines, so I rolled back to this.

Reply:
I'm good with that. No point in being rigid just for the sake of 'style'. This looks fine :+1: